pkg/api/v1beta1/types.go

```diff
@@ -1522,3 +1526,27 @@ type GlusterfsVolumeSource struct {
 	// the Glusterfs volume to be mounted with read-only permissions
 	ReadOnly bool `json:"readOnly,omitempty" description:"Glusterfs volume to be mounted with read-only permissions"`
 }
+
+// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
+type RBDVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
+	EndpointsName string `json:"endpoints" description:"a collection of Ceph monitors"`
```

Comment:
So at a minimum so far it sounds like we should have used an object reference.  If we did that, and stated that we will for a period of time support endpoints as the Kind while we solve this, does that reduce the concern about API stability?  

> On Apr 13, 2015, at 2:46 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/api/v1beta1/types.go:
> 
> > @@ -1522,3 +1526,27 @@ type GlusterfsVolumeSource struct {
> >  	// the Glusterfs volume to be mounted with read-only permissions
> >  	ReadOnly bool `json:"readOnly,omitempty" description:"Glusterfs volume to be mounted with read-only permissions"`
> >  }
> > +
> > +// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
> > +type RBDVolumeSource struct {
> > +	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
> > +	EndpointsName string `json:"endpoints" description:"a collection of Ceph monitors"`
> In general I think a headless service is preferred to raw Endpoints, even if just to prevent collisions. But the discussion never concluded. My last comment was essentially "As written, the endpoints must be in the same namespace as the pod. I don't think this is what you want to do." We need to rethink how we want to describe connecting to these sorts of services, and I don't think we should commit this (and frankly neither gluster) until we have an asnwer. Gluster will now have to be backwards compatible or else break compat.
> 
> —
> Reply to this email directly or view it on GitHub.
>